account_journal_restrict_mode/models/account_journal.py

```diff
@@ -0,0 +1,29 @@
+# Copyright 2023 ForgeFlow S.L.
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
+
+from odoo import _, api, fields, models
+from odoo.exceptions import UserError
+
+
+class AccountJournal(models.Model):
+    _inherit = "account.journal"
+
+    restrict_mode_hash_table = fields.Boolean(default=True)
```

Comment:
```suggestion
    restrict_mode_hash_table = fields.Boolean(default=True, readonly=True)
```
Maybe we should also enforce readonly True to make sure it is not modified

Reply:
@JordiMForgeFlow Done! :)